### DISCUSSION TITLE
getUserObject in stream

### DISCUSSION THREAD

**Discussion by develocode777** [serverpod/serverpod#2046] [Q&A]
Is it possible to get the object set with `setUserObject` in `streamOpened` in a custom method in the endpoint? I noticed that session is a different type and maybe that's why I get null.

**Comment by SandPod** [comment]
That is correct, the endpoint and the streaming connection have different implementations of Session (`StreamingSession`, `Session`). This might be why you can't retrieve the same user object when fetching it from an endpoint.

What would you like to do with the user object if you could retrieve it from the endpoint method?

**Reply by develocode777** [reply]
I am doing some additional authentication both in the custom method and in `streamOpened` and save the info and I would like to do it only once.
